controllers/ovndbcluster_controller.go

```diff
@@ -576,11 +606,78 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 				instance,
 				svcLabels,
 			)
+			if err != nil {
+				err = fmt.Errorf("Error while deleting service with name %s: %w", serviceName, err)
+				return ctrl.Result{}, err
+			}
+			// Delete DNS records for deleted services/pods
+			namespace := helper.GetBeforeObject().GetNamespace()
+			dnsName := fmt.Sprintf("%s-%s-%d", v1beta1.DNSPrefix, serviceName, i)
```

Comment:
nit: the serviceName-%d is already constructed in line 601. you could construct it once, then add dns- here.

Reply:
DNS prefix was removed as asked by yatin. Done